configure.ac

```
@@ -266,6 +267,11 @@ AC_ARG_ENABLE([ocamldoc],
   [],
   [ocamldoc=auto])
 
+AC_ARG_WITH([odoc],
+  [AS_HELP_STRING([--with-odoc],
+    [build documentation with odoc])])
+
+
```

All `--with` flags have an argument - in this case, it might be useful to support it properly to allow `--with-doc ~/odoc-dev/_build/...`

```suggestion
documentation_tool_cmd=''
AC_ARG_WITH([odoc],
  [AS_HELP_STRING([--with-odoc],
  [AS_CASE([$withval],
    [yes],[documentation_tool='odoc'],
    [no],[documentation_tool='ocamldoc'],
    [documentation_tool_cmd="$withval"
    documentation_tool='odoc'])],
  [documentation_tool='ocamldoc'])

AS_IF([test "x$documentation_tool_cmd" = 'x'],[documentation_tool_cmd="$documentation_tool"])
```
(the `[no]` case is what autoconf does if you explicitly say `--without-odoc` vs the final parameter to `AC_ARG_WITH` which is what's used if you don't specify either `--with-odoc` or `--without-odoc`)

---

Updated.